src/atoms/TextInput/TextInput.web.js

```diff
@@ -1,3 +1,508 @@
+    prefix,
+    suffix,
+    iconLeft,
+    iconRight,
+  });
+
+  if (labelPosition === 'left' && variant === 'outlined') {
+    // Outlined Text Input has only a top label
+    throw Error('Cannot have a left label on an outlined Text Input');
+  }
+
+  if (!isEmpty(prefix) && !isEmpty(iconLeft)) {
+    throw Error('Cannot have prefix and left icon together');
+  }
+
+  if (!isEmpty(suffix) && !isEmpty(iconRight)) {
+    throw Error('Cannot have suffix and right icon together');
+  }
+
+  const onFocus = useCallback(() => {
+    setIsFocused(true);
+    /* Wait for 90ms to show the placeholder since it takes 100ms for Label to animate from inside to top of the TextInput.
+       Otherwise they both overlap */
+    /* Don't have any delay if label is on left of TextInput */
+
+    setTimeout(() => {
+      setIsPlaceholderVisible(true);
+    }, 90);
+  }, [setIsFocused, setIsPlaceholderVisible]);
+
+  const onBlurText = useCallback(
+    (event) => {
+      setIsFocused(false);
+      setIsPlaceholderVisible(false);
+      const inputValue = event.target.value;
+      setInput(inputValue);
+      if (onBlur) {
+        onBlur(inputValue);
+      }
+    },
+    [setIsFocused, setIsPlaceholderVisible, setInput, onBlur],
+  );
+
+  const onChangeText = useCallback(
+    (event) => {
+      const inputValue = event.target.value;
+      if (inputValue.length > maxLength) {
+        return;
+      }
+      setInput(inputValue);
+      if (onChange) {
+        onChange(inputValue);
+      }
+    },
+    [maxLength, onChange, setInput],
+  );
+
+  const onKeyPress = useCallback(
+    (event) => {
+      if (type === 'number') {
+        const charCode = typeof event.which === 'number' ? event.which : event.keyCode;
+        const char = String.fromCharCode(charCode);
+        const isAllowed = /[0-9]/g.test(char) || char === '-' || char === '.';
+
+        if (!isAllowed) {
+          event.preventDefault();
+        }
+      }
+    },
+    [type],
+  );
+
+  useEffect(() => {
+    if (value) {
+      setInput(value);
+    }
+  }, [value]);
+
+  useEffect(() => {
+    if (inputRef.current && containerRef.current) {
+      const containerRect = containerRef.current.getBoundingClientRect();
+      const inputRect = inputRef.current.getBoundingClientRect();
+      const initialLeftPosition = inputRect.x - containerRect.x;
+      const finalTopPosition = inputRef.current.offsetHeight;
+
+      setLayoutDimensions({
+        initialLeftPosition,
+        finalTopPosition,
+      });
+    }
+  }, [inputRef, containerRef, setLayoutDimensions]);
+
+  useEffect(() => {
+    // adjust height of textarea as user types for outlined variant
+    if (_isMultiline && variant === 'outlined') {
+      inputRef.current.style.height = 'inherit';
+      const height = inputRef.current.scrollHeight;
+      inputRef.current.style.height = `${height}px`;
+    }
+  }, [_isMultiline, variant, inputRef, input]);
+
+  const _rows = variant === 'outlined' ? 1 : 3;
```

Comment:
```suggestion
  const noOfRows = variant === 'outlined' ? 1 : 3;
```

Reply:
Let's follow to use `_` only for internal "props". Not variables. We're not using `_` for other internal variables anyways.